barretenberg/cpp/src/barretenberg/dsl/acir_format/acir_format.hpp

```diff
@@ -24,6 +24,44 @@
 
 namespace acir_format {
 
+/**
+ * @brief Indices of the original opcode that originated each constraint in AcirFormat.
+ * @details Contains one array of indices per opcode type. The length of each array is equal to the number of
+ * constraints of that type. The relationship between the opcodes and constraints is assumed to be one to one, except
+ * for block constraints.
+ */
+struct AcirFormatOriginalOpcodeIndices {
+    std::vector<size_t> logic_constraints;
+    std::vector<size_t> range_constraints;
+    std::vector<size_t> aes128_constraints;
+    std::vector<size_t> sha256_constraints;
+    std::vector<size_t> sha256_compression;
+    std::vector<size_t> schnorr_constraints;
+    std::vector<size_t> ecdsa_k1_constraints;
+    std::vector<size_t> ecdsa_r1_constraints;
+    std::vector<size_t> blake2s_constraints;
+    std::vector<size_t> blake3_constraints;
+    std::vector<size_t> keccak_constraints;
+    std::vector<size_t> keccak_permutations;
+    std::vector<size_t> pedersen_constraints;
+    std::vector<size_t> pedersen_hash_constraints;
+    std::vector<size_t> poseidon2_constraints;
+    std::vector<size_t> multi_scalar_mul_constraints;
+    std::vector<size_t> ec_add_constraints;
+    std::vector<size_t> recursion_constraints;
+    std::vector<size_t> honk_recursion_constraints;
+    std::vector<size_t> bigint_from_le_bytes_constraints;
+    std::vector<size_t> bigint_to_le_bytes_constraints;
+    std::vector<size_t> bigint_operations;
+    std::vector<size_t> poly_triple_constraints;
+    std::vector<size_t> quad_constraints;
+    // Multiple opcode indices per block:
+    std::vector<std::vector<size_t>> block_constraints;
+
+    friend bool operator==(AcirFormatOriginalOpcodeIndices const& lhs,
```

Comment:
Do you need to be explicit on == ?

Reply:
If I didn't do this the parent struct complains about `Explicitly defaulted equality comparison operator is implicitly deletedclang(-Wdefaulted-function-deleted)
`

Reply:
IDK why the parent is explicit though